[Feat] Add TORCH_DISTRIBUTED_BACKEND env variable (#5981)
PyTorchLightning/pytorch-lightning: docs/source/advanced/multi_gpu.rst

    to be in "exclusive mode", such that only one process at a time can access them.
    For more details see the :doc:`trainer guide <../common/trainer>`.

> Select Torch Distributed Backend

Reviewer: the other titles don't use capitalization

Author: removed !

--------------------------------

> By default, Lightning will choose `nccl` backend over `gloo` when running on GPUs.

Reviewer 2: ```suggestion
By default, Lightning will select the ``nccl`` backend over ``gloo`` when running on GPUs.
```

Author: Done !

> Find more information on [PyTorch supported backends](https://pytorch.org/docs/stable/distributed.html).

Reviewer 2: ```suggestion
Find more information about PyTorch's supported backends `here <https://pytorch.org/docs/stable/distributed.html>`_.
```

Author: Done !

> Find more information about PyTorch's supported backends `here <https://pytorch.org/docs/stable/distributed.html>`_.

Reviewer 2: ```suggestion
Find more information about PyTorch's supported backends `here <https://pytorch.org/docs/stable/distributed.html>`__.
```

> Lightning exposes an environment variable ``TORCH_DISTRIBUTED_BACKEND`` for the user to change the backend.

Reviewer: aren't all env variables in pl prefixed with PL?

Author: Added prefix.
